apps/web/src/app/(public)/sign-in/_components/wizard/context.tsx

```diff
@@ -47,6 +56,20 @@ export const SignInWizardProvider = ({
   const [vfnStepData, setVfnStepData] = useState<VfnStepData>()
   const [timer, setTimer] = useState(delayForResendSeconds)
 
+  const challengeToVerifierMap = useRef(new Map<string, string>())
+  const newChallenge = async () => {
+    const verifier = browserCreatePkceVerifier()
+    const challenge = await browserCreatePkceChallenge(verifier)
```

Comment:
this function might throw, do you want to add some error handling using try-catch here? Just a toast should do (there is already a toast component)